NET21Solution/NET021Task/Catalog.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NET021Task
+{
+    public class Catalog : IEnumerable<Book>
+    {
+        public List<Book> Books { get; private set; }
+
+        public Catalog()
+        {
+            Books = new List<Book>();
+        }
+
+        public Book this[string isbn]
+        {
+            get
+            {
+                if (!Book.ISBNPattern1.IsMatch(isbn))
```

Comment:
My idea was to only allow one ISBN format to access books in catalog (since in the Book class the format is normalized in the ctor), so I'm only checking if the supplied ISBN format is not the normalized one...